testing-internal/src/main/java/com/linecorp/centraldogma/testing/internal/ExpectedExceptionAppender.java

```diff
@@ -0,0 +1,269 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.testing.internal;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.slf4j.Marker;
+
+import com.linecorp.armeria.common.util.SafeCloseable;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.classic.spi.LoggerContextVO;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import ch.qos.logback.core.spi.AppenderAttachable;
+import ch.qos.logback.core.spi.AppenderAttachableImpl;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+import io.netty.util.internal.logging.Slf4JLoggerFactory;
+
+/**
+ * A <a href="https://logback.qos.ch/">Logback</a> {@link Appender} that adds {@code "(expected exception)"}
+ * to the log messages of expected exceptions and lower their logging level to {@link Level#DEBUG}.
+ */
+public class ExpectedExceptionAppender extends UnsynchronizedAppenderBase<ILoggingEvent>
+        implements AppenderAttachable<ILoggingEvent> {
+
+    private static final String EXPECTED_EXCEPTION = "(expected exception)";
+
+    private static final ConcurrentMap<String, String> exceptionAndCause = new ConcurrentHashMap<>();
+
+    static {
+        if (InternalLoggerFactory.getDefaultFactory() == null) {
+            // Can happen due to initialization order.
+            InternalLoggerFactory.setDefaultFactory(Slf4JLoggerFactory.INSTANCE);
+        }
+    }
+
+    private final AppenderAttachableImpl<ILoggingEvent> aai = new AppenderAttachableImpl<>();
+
+    /**
+     * Adds {@code "(expected exception)"} to the log message and lower its logging level to
+     * {@link Level#DEBUG} if the log's exception matches {@code throwable}, and contains {@code cause}.
+     * You need to use this to only specific range that occurs exception you expected using a
+     * {@code try-with-resources} block:
+     * <pre>{@code
+     * try (SafeCloseable ignored = ExpectedExceptionAppender.appendIfMatch(throwable, cause)) {
+     *     ...
+     *     // Expected exception happens.
+     *     ...
+     * }
+     * }</pre>
+     */
+    public static SafeCloseable appendIfMatch(Class<?> throwable, String cause) {
```

Comment:
`appendIfMatch` makes me feel like the event will not be appended if not matches. How about: `withExpectedException()`? This also makes me think the following pattern may be better because `SafeCloseable` is unnecessary completely.

```java
withExpectedException(SomeException.class, "text", () -> {
    assertThatThrownBy(...);
});
```

Reply:
That's a good idea~!! Thank you~!